### ISSUE TITLE
Poco f1 windows install Tutorial please

### ISSUE THREAD

**Issue by hwanghyunseo** [edk2-porting/edk2-sdm845#51]
There is no way to install Windows arm os on poco f1 on YouTube.  There is only a video in Portuguese, but it is an introduction video, not a method.  Please tell me how to install on pocophone f1.

**Comment by hwanghyunseo** [comment]
Portuguese video link:https://youtu.be/t81-Ji17fGM

**Comment by salam-siddique** [comment]
installed successfully on 2 poco f1 phones

**Comment by hwanghyunseo** [comment]
> installed successfully on 2 poco f1 phones
How to?

**Comment by Tonio1987fr** [comment]
The guide for the oneplus 6 works. You just have to modify the partitions for the Pocophone.

Attention the touchscreen does not work on the Pocophone f1 =(

**Comment by neozeed** [comment]
> installed successfully on 2 poco f1 phones

What version of Windows with which bootloader?
Release - 2301.1 for the ebbg doesn't boot windows at all, but it seems older ones do?  But I always crash with MEMORY_PROTECTION or IRQL less than expected.

**Comment by ComitC3** [comment]
Hey, it's because of some driver issues(maybe) I have used some old drivers and removed one. And now all the thing works(that's supposed to work). Touch also works. I also get to use that error but now everything works as stated. For fix Drivers- use usb drivers for rc02, not latest, and then if your device is FTS panel, remove the nt36xx folder of driver and only keep focaltouch folder.total there will be 61 drivers instead of 62. Also ebbg devices might have to remove pmic folder also for some devices.use the given tutorial in the installation page , its enough.

**Comment by jukfiuune** [comment]
There is literally a tutorial on the website. USB and touchscreen work on my device. Only thing that I know doesn't work is the latest Windows 11 Canary and Dev build. Here: https://renegade-project.tech/en/devices/xiaomi/beryllium

**Comment by neozeed** [comment]
I used 10.0.21390.1.co_release.210521-1658_arm64fre for the windows install. useing the [following link](https://uupdump.net/download.php?id=6db21a1e-53af-48cb-a0ba-94e1b852c33b&pack=en-us&edition=professional) to make sure it got the updates etc.

I assume the drivers in question is [v2.0rc2](https://github.com/edk2-porting/WOA-Drivers/releases/download/v2.0rc2/beryllium.tar.gz)

I added the drivers, made sure to Enable test signing and disable Automatic Repair

I used the bootloader: boot-beryllium-ebbg Release - 2210.1.img as 23xx doesnt boot anything.

and blinking like crazy 
![could not](https://github.com/edk2-porting/edk2-msm/assets/9031439/373f38b5-b139-4c70-a4be-f95cd95fc374)

**Comment by ComitC3** [comment]
I think I should try the way I stated. It might work. Worked for me at least I too couldn't boot in latest 2301 img. If ebbg driver, u need to try removing the pmic driver folder before installing drivers

**Comment by neozeed** [comment]
![10 0 21390 1 co_release 210521-1658_arm64fre beryllium v2 0rc2 no PMIC](https://github.com/edk2-porting/edk2-msm/assets/9031439/15927895-6340-43e8-a00f-d10b9b0091df)

So here is how I got here...

Prep the disk, format etc etc.. nothing too special?
```
C:\Users\neozeed\Downloads\adb>diskpart

Microsoft DiskPart version 10.0.19041.964

Copyright (C) Microsoft Corporation.
On computer: ERYING

DISKPART> list disk

  Disk ###  Status         Size     Free     Dyn  Gpt
  --------  -------------  -------  -------  ---  ---
  Disk 0    Online          953 GB  1024 KB        *
  Disk 1    Online          953 GB  1024 KB        *
  Disk 2    Online         1863 GB      0 B
  Disk 3    Offline          32 MB  8128 KB        *
  Disk 4    Offline          32 MB    28 MB        *
  Disk 5    Offline           8 MB   384 KB        *
  Disk 6    Offline        4608 MB    63 MB        *
  Disk 7    Offline           8 MB   384 KB        *
  Disk 8    Offline         114 GB      0 B        *

DISKPART> select disk 8

Disk 8 is now the selected disk.

DISKPART> online

Microsoft DiskPart version 10.0.19041.964

DISK        - Online a disk that is currently marked as offline.
VOLUME      - Online a volume that is currently marked as offline.

DISKPART> online disk

DiskPart successfully onlined the selected disk.

DISKPART> list volume

  Volume ###  Ltr  Label        Fs     Type        Size     Status     Info
  ----------  ---  -----------  -----  ----------  -------  ---------  --------
  Volume 0     C                NTFS   Partition    953 GB  Healthy    Boot
  Volume 1                      FAT32  Partition    100 MB  Healthy    System
  Volume 2                      NTFS   Partition    530 MB  Healthy    Hidden
  Volume 3     D   Data         NTFS   Partition    953 GB  Healthy
  Volume 4     U   TOSHIBA EXT  NTFS   Partition   1863 GB  Healthy
  Volume 5         SYSTEM       FAT32  Partition    476 MB  Healthy    Hidden
  Volume 6         Windows      NTFS   Partition     84 GB  Healthy

DISKPART> select volume 5

Volume 5 is the selected volume.

DISKPART> format quick fs=fat32 label="System"

  100 percent completed

DiskPart successfully formatted the volume.

DISKPART> assign letter="S"

DiskPart successfully assigned the drive letter or mount point.

DISKPART> select volume 6

Volume 6 is the selected volume.

DISKPART> format quick fs=ntfs label="Windows"

  100 percent completed

DiskPart successfully formatted the volume.

DISKPART> assign letter="W"

DiskPart successfully assigned the drive letter or mount point.

DISKPART> exit

Leaving DiskPart...

```

Next load the image:

```
C:\Users\neozeed\Downloads\adb>dism /apply-image /ImageFile:10.0.21390.1.co_release.210521-1658_arm64fre\install.wim /index:1 /ApplyDir:W:\

Deployment Image Servicing and Management tool
Version: 10.0.19041.844

Applying image
[==========================100.0%==========================]
The operation completed successfully.

C:\Users\neozeed\Downloads\adb>


C:\Users\neozeed\Downloads\adb>bcdboot W:\Windows /s S: /f UEFI
Boot files successfully created.
```

Make sure there is no PMIC and apply the drivers:
```
C:\Users\neozeed\Downloads\adb>dir "WOA-Drivers\beryllium v2.0rc2"\*pmic* /s
 Volume in drive C has no label.
 Volume Serial Number is B246-CD96
File Not Found






C:\Users\neozeed\Downloads\adb>dism /Image:W: /Add-Driver /Driver:"WOA-Drivers\beryllium v2.0rc2" /Recurse

Deployment Image Servicing and Management tool
Version: 10.0.19041.844

Image Version: 10.0.21390.1

Searching for driver packages to install...
Found 58 driver package(s) to install.
Installing 1 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\10420\qcsmmu850.inf: The driver package was successfully installed.
Installing 2 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\10730\qciommu850.inf: The driver package was successfully installed.
Installing 3 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\ACPI\qcabd850.inf: The driver package was successfully installed.
Installing 4 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Bluetooth\AddVSCRegistry\qcbtaddvscregistry850.inf: The driver package was successfully installed.
Installing 5 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Bluetooth\UART\qcbtfmuart850.inf: The driver package was successfully installed.
Installing 6 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Bluetooth\UART_ext\qcbtfmuart_ext850.inf: The driver package was successfully installed.
Installing 7 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\BUS\qcgpi850.inf: The driver package was successfully installed.
Installing 8 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\BUS\qci2c850.inf: The driver package was successfully installed.
Installing 9 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\BUS\qcspi850.inf: The driver package was successfully installed.
Installing 10 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\BUS\qcuart850.inf: The driver package was successfully installed.
Installing 11 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\Databus\qcdatabus850.inf: The driver package was successfully installed.
Installing 12 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\DiagBridge\QCDiagBridge850.inf: The driver package was successfully installed.
Installing 13 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\DiagCSI\qcdiagcsi850.inf: The driver package was successfully installed.
Installing 14 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\DiagRouter\qcdiagrouter850.inf: The driver package was successfully installed.
Installing 15 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\DplBridge\qcdplbridge850.inf: The driver package was successfully installed.
Installing 16 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\IPA\qcgsi850.inf: The driver package was successfully installed.
Installing 17 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\IPA\qcipa850.inf: The driver package was successfully installed.
Installing 18 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\IPCRouter\qcipcrouter850.inf: The driver package was successfully installed.
Installing 19 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\Linklocal\qclinklocal.wp850.inf: The driver package was successfully installed.
Installing 20 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\LteCoexistanceManager\QcLteCoexMgr850.inf: The driver package was successfully installed.
Installing 21 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\MBB\qcmbb850.inf: The driver package was successfully installed.
Installing 22 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\MBRG\qcmbrg850.inf: The driver package was successfully installed.
Installing 23 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\RemoteAt\qcremoteat850.inf: The driver package was successfully installed.
Installing 24 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\RemoteFS\qcremotefs850.inf: The driver package was successfully installed.
Installing 25 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\revrmnet\qcrevrmnet850.inf: The driver package was successfully installed.
Installing 26 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Cellular\RmnetBridge\qcrmnetbridge850.inf: The driver package was successfully installed.
Installing 27 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\FG\BattMgr\qcbattmngr850.inf: The driver package was successfully installed.
Installing 28 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\FG\BattMiniclass\qcbattminiclass850.inf: The driver package was successfully installed.
Installing 29 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\FG\FGBCL\qcfgbcl850.inf: The driver package was successfully installed.
Installing 30 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\FocalTechTouch\FocalTechTouch.inf: The driver package was successfully installed.
Installing 31 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\GLINK\qcglink850.inf: The driver package was successfully installed.
Installing 32 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\GPIO\qcgpio850.inf: The driver package was successfully installed.
Installing 33 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\GPS\qcgnss850.inf: The driver package was successfully installed.
Installing 34 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\GPU\qcdx850.inf: The driver package was successfully installed.
Installing 35 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\HALs\HalExtQCWdogTimer850.inf: The driver package was successfully installed.
Installing 36 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\HexagonLoader\qcpil850.inf: The driver package was successfully installed.
Installing 37 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\nt36xxx\nt36xxx.inf: The driver package was successfully installed.
Installing 38 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\PDSR\qcpdsr850.inf: The driver package was successfully installed.
Installing 39 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\PEP\qcpep.wd850.inf: The driver package was successfully installed.
Installing 40 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\qSarMgr\qSarMgr850.inf: The driver package was successfully installed.
Installing 41 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\ResetPower\qccdi850.inf: The driver package was successfully installed.
Installing 42 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\ResetPower\qcrpen850.inf: The driver package was successfully installed.
Installing 43 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\RNG\qcRng850.inf: The driver package was successfully installed.
Installing 44 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\RPC\qcadsprpc850.inf: The driver package was successfully installed.
Installing 45 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\SCM\qcscm850.inf: The driver package was successfully installed.
Installing 46 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\SPMI\qcspmi850.inf: The driver package was successfully installed.
Installing 47 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\SSVC\qcshutdownsvc850.inf: The driver package was successfully installed.
Installing 48 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Subsys_Extension\qcsubsys_extension850.inf: The driver package was successfully installed.
Installing 49 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Subsystems\MCFG_Subsys_ext\mcfg_subsys_ext850.inf: The driver package was successfully installed.
Installing 50 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Subsystems\SSDD\qcsubsys850.inf: The driver package was successfully installed.
Installing 51 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\System\qckmbam850.inf: The driver package was successfully installed.
Installing 52 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\System\qcsyscache850.inf: The driver package was successfully installed.
Installing 53 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\Tftp\QcTftpKmdf850.inf: The driver package was successfully installed.
Installing 54 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\TrEE\QcTrEE850.inf: The driver package was successfully installed.
Installing 55 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\USB\USBC\qcusbc850.inf: The driver package was successfully installed.
Installing 56 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\USB\XhciFilter\QcXhciFilter850.inf: The driver package was successfully installed.
Installing 57 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\VoltageCurrentMonitoring\qcadc850.inf: The driver package was successfully installed.
Installing 58 of 58 - C:\Users\neozeed\Downloads\adb\WOA-Drivers\beryllium v2.0rc2\WLAN\qcwlan850.inf: The driver package was successfully installed.
The operation completed successfully.
```

update boot variables
```
C:\Users\neozeed\Downloads\adb>cd S:\EFI\Microsoft\Boot

C:\Users\neozeed\Downloads\adb>s:

S:\EFI\Microsoft\Boot>
S:\EFI\Microsoft\Boot>bcdedit /store BCD /set "{default}" testsigning on
The operation completed successfully.

S:\EFI\Microsoft\Boot>bcdedit /store BCD /set "{default}" nointegritychecks on
The operation completed successfully.

S:\EFI\Microsoft\Boot>bcdedit /store BCD /set "{default}" recoveryenabled no
The operation completed successfully.
```

Then I remove the disk letters, unmount the disk reboot.. and yeah.
```
C:\Users\neozeed\Downloads\adb>diskpart

Microsoft DiskPart version 10.0.19041.964

Copyright (C) Microsoft Corporation.
On computer: ERYING

DISKPART> list volume

  Volume ###  Ltr  Label        Fs     Type        Size     Status     Info
  ----------  ---  -----------  -----  ----------  -------  ---------  --------
  Volume 0     C                NTFS   Partition    953 GB  Healthy    Boot
  Volume 1                      FAT32  Partition    100 MB  Healthy    System
  Volume 2                      NTFS   Partition    530 MB  Healthy    Hidden
  Volume 3     D   Data         NTFS   Partition    953 GB  Healthy
  Volume 4     U   TOSHIBA EXT  NTFS   Partition   1863 GB  Healthy
  Volume 5     W   Windows      NTFS   Partition     84 GB  Healthy
  Volume 6     S   SYSTEM       FAT32  Partition    476 MB  Healthy    Hidden

DISKPART> select volume s

Volume 6 is the selected volume.

DISKPART> remove letter=s

DiskPart successfully removed the drive letter or mount point.

DISKPART> select volume 5

Volume 5 is the selected volume.

DISKPART> remove letter=w

DiskPart successfully removed the drive letter or mount point.

DISKPART> offline disk

DiskPart successfully offlined the selected disk.

DISKPART> exit

Leaving DiskPart...

C:\Users\neozeed\Downloads\adb>
```

I'm pretty sure it must be the combination of what version of windows vs trying to load drivers?

**Comment by neozeed** [comment]
Okay some success!

In that Brazilian video he's using 210521-1658 so its's def a known good version.  In the 2210.1-fix drivers I saw there is a minimal directory so I applied that, while swapping the USB from beryllium v2.0rc2.  I also ran the 'touchfixup', and of course all the BCD flags.  I did add this flag:

`bcdedit /store BCD /set "{default}" debug on`

Oddly enough the 2301.1 bootloader work!?  First boot it does the getting ready and reboots.  Second boot it went into the setup steps, and I got a OTG adapter, plugged in a mouse and completed the setup.

Touchscreen didn't work, and there was no networking.  But it did boot!


```
10.0.21390.1.co_release.210521-1658_arm64fre
boot-beryllium-ebbg Release - 2301.1.img	2210.1-fix\minimal	getting ready	reboot	
									no usb no touch

```
